File: DataBinding/PageAndRDLX/ODataDataSource/C#/JsonDataSourceClient/DataLayer.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using System;
using System.Net;
using System.Net.Http;
using System.Text;

namespace ActiveReports.Samples.ODataDataSource
{
	internal class DataLayer
	{
		private static readonly string[] Sources = 
			{
			@"http://localhost:55749/Customers?$format=application/json;odata.metadata=none",
			@"https://services.odata.org/V4/Northwind/Northwind.svc/Customers?$select=CustomerID,%20CompanyName,%20ContactName,%20Address%20&%20$format=application/json;odata.metadata=none"
			};

		public static string CreateData(Service service)
		{
			var sourceUrl = Sources[(int)service];

			using (var httpClient = new HttpClient())
			{
				var request = new HttpRequestMessage(HttpMethod.Get, sourceUrl);
				var response = httpClient.SendAsync(request).Result;
				var json = response.Content.ReadAsStringAsync().Result;
				var jObject = (JObject)JsonConvert.DeserializeObject(json);
				foreach (var obj in jObject)
				{
					if (obj.Key == "value")
					{
						return "{\"" + obj.Key + "\":" + obj.Value + "}";
					}
				}
				return "";
			}
		}
	}
}

File: DataBinding/PageAndRDLX/ODataDataSource/VB.NET/JsonDataSourceClient/DataLayer.vb
Imports System.Net.Http
Imports Newtonsoft.Json.Linq
Imports Newtonsoft.Json

Friend Class DataLayer

	Private Shared ReadOnly Sources As String() = {
		"http://localhost:55856/Customers?$format=application/json;odata.metadata=none",
		"https://services.odata.org/V4/Northwind/Northwind.svc/Customers?$select=CustomerID,%20CompanyName,%20ContactName,%20Address%20&%20$format=application/json;odata.metadata=none"
	}

	Public Shared Function CreateData(ByVal service As Service) As String
		Dim sourceUrl = Sources(CInt(service))

		Using httpClient = New HttpClient()
			Dim request = New HttpRequestMessage(HttpMethod.[Get], sourceUrl)
			Dim response = httpClient.SendAsync(request).Result
			Dim json = response.Content.ReadAsStringAsync().Result
			Dim jObject = CType(JsonConvert.DeserializeObject(json), JObject)

			For Each obj In jObject

				If obj.Key = "value" Then
					Return "{""" & obj.Key & """:" & obj.Value.ToString() & "}"
				End If
			Next

			Return ""
		End Using
	End Function
End Class
